dds_web/crypt/auth.py

```diff
@@ -2,10 +2,39 @@
 
 import argon2
 
-from sqlalchemy.exc import SQLAlchemyError
+import sqlalchemy
 
 from dds_web.database import models
 from sqlalchemy.sql import func
+from dds_web import exceptions
+from dds_web 
+
+
+def verify_user_pass(username, password):
+    """Verify that user exists and password is correct."""
+
+    # Verify existing user
+    try:
+        user = models.User.query.filter(models.User.username == func.binary(username)).first()
+    except sqlalchemy.exc.SQLAlchemyError:
+        raise
+
+    # User does not exist
+    if not user:
+        raise exceptions.AuthenticationError("Incorrect username and/or password!")
+
+    # Verify password and generate token
+    if verify_password_argon2id(user.password, password):
+        if "l" not in list(user.permissions):
```

Comment:
Does this (if block for permissions) belong here? Permission is related to authorization. If this code piece is related to verifying the authenticity of a user, then by single responsibility principle it would be a better to move the authorization check to elsewhere.

Reply:
No, it doesn't. I had actually written a comment about that but I seem to have removed it again... will comment it out for now and move it somewhere else later.

Reply:
Fixed.